self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryContracts.java

```diff
@@ -53,12 +56,101 @@
 
     /**
      * Constructor.
+     *
      * @param storage Parent storage
      */
     public InMemoryContracts(final Storage storage) {
         this.storage = storage;
     }
 
+
+    /**
+     * Adds a contract based on valid projectId and contributorId.
+     * If either one of ids is invalid, an exception will be thrown.
+     *
+     * @param projectId Valid project id
+     * @param contributorId Valid contributor id
+     * @param hourlyRate Contract's hourly rate
+     * @param role Contract's role
+     * @return Contract
+     * @checkstyle ParameterNumber (10 lines)
+     */
+    public Contract addContract(final int projectId,
```

Comment:
@criske Declare this method on the ``Contracts`` interface please. Then, in ``ProjectContracts`` and ``ContributorContracts`` just make it throw ``UnsupportedOperationException("Not yet implemented")``.

All public methods of any class should be declared in its interface.